### PR TITLE
feat: option to exclude specific methods

### LINKED ISSUE #9: Whitelist driver.Valuer interface
This issue is related to issue https://github.com/raeperd/recvcheck/issues/7 which has been fixed by PR https://github.com/raeperd/recvcheck/pull/8 in order to reduce the number of false positives in the default configuration of the linter.

The interfaces [driver.Valuer](https://pkg.go.dev/database/sql/driver#Valuer) and [sql.Scanner](https://pkg.go.dev/database/sql#Scanner) from the standard library conflict with each other and are commonly implemented by the same struct. This was already mentioned in this comment https://github.com/raeperd/recvcheck/issues/7#issuecomment-2468474729 but somehow was not implemented in the PR.

It would be nice to add driver.Valuer to the default exclusions.

### PR DESCRIPTION
**Pull request by ldez**
This is a draft of a proposal.

I used a glob-based expression, but it can be a regular expression or something else.

Related to https://github.com/raeperd/recvcheck/pull/8#issuecomment-2480381982
Fixes #9

### CODE REVIEW

File: analyzer.go [hunk 1]
```diff
@@ -114,3 +137,17 @@ type structType struct {
...
+		if r, ok := n.X.(*ast.Ident); !ok {
+			return r, false
+		}
+
+	case *ast.Ident:
+		return n, false
+	}
+
+	return nil, false
+}
```
Comment (ccoVeille) [inline analyzer.go:153]: Is it normal ypu always return false?

It looks like a strange signature

**Comment by raeperd** [comment]
To me, `*.MarshalYAML` or `MarshalYAML` does the same. 
So maybe we can safely remove glob pattern, and simplifies config with exact match only. (ex `type.MarshalYAML`)

**Comment by raeperd** [comment]
@ldez 
is there any thing more to be done in this pr? maybe I can help

**Comment by ldez** [comment]
I was just waiting for your feeback

**Comment by raeperd** [comment]
@ldez 
Sorry.. I read your comment right after your comment,  and add emoji to show you I agreed with your comment. I will start review now.

**Review by raeperd — CHANGES_REQUESTED** [review]

**Review by raeperd — APPROVED** [review]
lgtm

**Pull request merged**